File: jour_18/jour18.c
#include<stdio.h>
int main (){
    int taille,result,c;
    do {
        printf("Veuillez saisier la taille du tableau : ");
        result = scanf("%d",&taille);
        while((c = getchar()) != '\n' && c != EOF);
        if (result == 0)
            printf("Saisie invalid\n");
        else if (taille <= 0)
            printf("Veuillez entrer un nombre positif\n");
    }while(result == 0 || taille <= 0);
    int tab[taille];
    printf("Entrez la valeur de : \n");
    for (int i = 0; i < taille; i++){
        printf("tab[%d] = ",i);
        scanf("%d",&tab[i]);
    }
    for (int i = 0; i < taille; i++){
        int indMin = i;
        for (int j = i + 1; j < taille; j++){
            if (tab[j] < tab[indMin])
                indMin = j;
        } 
        int temp = tab[indMin];
        tab[indMin] = tab[i];
        tab[i] = temp;
        
    }
    printf("le tableau trié : \n");
    for (int i = 0; i < taille ; i++){
        printf("tab[%d] = %d\n",i,tab[i]);
    }
    return 0;
}

File: jour_18/jour18.py
import sys

while True:
    try:
        taille = int(input("Veuillez saisir la taille du tableau : "))
        if taille > 0:
            break
        print("Veuillez entrer un nombre positif")
    except ValueError:
        print("Saisie invalide")

tab = []
print("Entrez la valeur de : ")
for i in range(taille):
    tab.append(int(input(f"tab[{i}] = ")))

for i in range(taille):
    indMin = i
    for j in range(i + 1, taille):
        if tab[j] < tab[indMin]:
            indMin = j
    
    tab[i], tab[indMin] = tab[indMin], tab[i]

print("le tableau trié : ")
for i in range(taille):
    print(f"tab[{i}] = {tab[i]}")